src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -868,14 +869,22 @@ private void VerifyCoresPerNode(Dictionary<string, DetailsPerNode> detailsPerNod
 
         private void SetAffinity(Process process, int cores, bool addPerformanceHint)
         {
+            if (cores > ProcessorInfo.ProcessorCount)
+            {
+                // the number of assigned cores is larger than we have on this machine
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Cannot set affinity to {cores} core(s), " +
+                                $"when the number of cores on this machine is: {ProcessorInfo.ProcessorCount}");
+                return;
+            }
             try
             {
                 var currentlyAssignedCores = NumberOfSetBits(process.ProcessorAffinity.ToInt64());
-                if (cores > ProcessorInfo.ProcessorCount ||
-                    currentlyAssignedCores == ProcessorInfo.ProcessorCount)
+                if (currentlyAssignedCores == cores &&
+                    _lastPerformanceHint != null && 
+                    _lastPerformanceHint.Value.AddDays(7) > DateTime.UtcNow)
```

Comment:
Why do you have this check here?

Reply:
This runs every day as part of checking for an updated license.